Implement UpdateConfigRaw in hyperkit driver

cfergeau opened the pull request:
This will make it possible to update memory size/cpu count of the VM.

cfergeau committed:
Use CopyFile rather than CopyDiskToMachineDir

CopyDiskToMachineDir will assume that the URL we gave it is an actual
URL, and will call a method called 'Download'. Since all we want is to
copy the disk image from one location to another, we can directly call
the appropriate method and skip the indirection.
CopyDiskToMachineDir is gone in newer versions of the machine package,
and Driver.DiskPathURL is replaced by Driver.ImageSourcePath, this
commit adjusts to these changes.

cfergeau committed:
Implement UpdateConfigRaw for memory/vcpu changes

Since hyperkit does not have any persistent state, we don't need to
do anything special here, we just sync the driver state. The Start() method
will always reuse this state to create a VM with the right number of
CPUs/amount of memory/...

cfergeau committed:
Set driver version to 0.12.8

Incompatible changes were made to the machine driver, this will
ensure crc can pick up a machine-driver-hyperkit binary with the right
ABI.

cfergeau committed:
Favour d.ResolveStorePath() over filepath.Join()

The Driver type has a helper to resolve path relative to the state dir,
so it's better to use it rather than hardcoding that logic in the
machine driver as well.

praveenkumar commented:
Run `go mod tidy` should be part of https://github.com/code-ready/machine-driver-hyperkit/pull/19/commits/d3a63490379810344716f8b2090abc308b6f6e55 one where we are updating the machine version to go.mod instead of a separate commit message.

cfergeau commented:
> Run `go mod tidy` should be part of [d3a6349](https://github.com/code-ready/machine-driver-hyperkit/commit/d3a63490379810344716f8b2090abc308b6f6e55) one where we are updating the machine version to go.mod instead of a separate commit message.

Running `git mod tidy` on current master (before this patch series) generates a similar diff to the dedicated commit in this PR, so the go mod tidy change is not related to the other commits in this PR. I'll squash it as it's not so important

praveenkumar commented on pkg/hyperkit/driver.go:
```diff
@@ -404,3 +409,14 @@ func (d *Driver) getPid() int {
 
 	return config.Pid
 }
+func (d *Driver) UpdateConfigRaw(rawConfig []byte) error {
```
This shouldn't be part of https://github.com/code-ready/machine-driver-hyperkit/pull/19/commits/482ba186d3323b5130f26e1b76829063bdce7294 commit.

cfergeau committed:
Don't hardcode 'crc.disk' as the image name

Users of crc's machine package now provide the image format in addition
to the machine name, so we can automatically build a proper name from
this rather than hardcoding it.

praveenkumar commented:
https://github.com/code-ready/machine-driver-hyperkit/pull/19/commits/d1eea888da3be4a6567acdbf489ed0e25fd8f7aa logs says `Set driver version to 0.12.9` but it should be `0.12.8`

praveenkumar commented on pkg/hyperkit/constants.go:
```diff
@@ -2,7 +2,7 @@ package hyperkit
 const (
 	DriverName    = "hyperkit"
-	DriverVersion = "0.12.7"
+	DriverVersion = "0.12.8"
```
:+1:
